pkg/synopsysctl/cmd_create.go

```diff
@@ -648,172 +644,116 @@ var createOpsSightNativeCmd = &cobra.Command{
-		if err != nil {
-			cmd.Help()
-			return err
+			return fmt.Errorf("this command takes 0 argument, but got %+v", args)
 		}
-		cobra.MarkFlagRequired(cmd.Flags(), "version")
-		cobra.MarkFlagRequired(cmd.Flags(), "environment-dns")
-		cobra.MarkFlagRequired(cmd.Flags(), "postgres-password")
-
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-host")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-port")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-username")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-password")
-		cobra.MarkFlagRequired(cmd.Flags(), "smtp-sender-email")
-
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-description")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-name")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-username")
-		cobra.MarkFlagRequired(cmd.Flags(), "organization-admin-email")
-		cobra.MarkFlagRequired(cmd.Flags(), "polaris-license-path")
-		cobra.MarkFlagRequired(cmd.Flags(), "coverity-license-path")
-		cobra.MarkFlagRequired(cmd.Flags(), "gcp-service-account-path")
-
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		polarisObj, err := updatePolarisSpecWithFlags(cmd, namespace)
+		// Get the flags to set Helm values
+		helmValuesMap, err := createPolarisCobraHelper.GenerateHelmFlagsFromCobraFlags(cmd.Flags())
 		if err != nil {
 			return err
 		}
 
-		if err := CheckVersionExists(baseURL, polarisObj.Version); err != nil {
-			return err
+		// Update the Helm Chart Location
+		chartLocationFlag := cmd.Flag("chart-location-path")
+		if chartLocationFlag.Changed {
+			polarisReportingChartRepository = chartLocationFlag.Value.String()
+		} else {
+			versionFlag := cmd.Flag("version")
+			if versionFlag.Changed {
+				polarisReportingChartRepository = fmt.Sprintf("%s/charts/polaris-helmchart-reporting-%s.tgz", baseChartRepository, versionFlag.Value.String())
+			}
 		}
 
-		components, err := polaris.GetComponents(baseURL, *polarisObj)
+		// Get Secret For the GCP Key
+		gcpServiceAccountPath := cmd.Flag("gcp-service-account-path").Value.String()
+		gcpServiceAccountData, err := util.ReadFileData(gcpServiceAccountPath)
 		if err != nil {
-			return err
+			return fmt.Errorf("failed to read gcp service account file at location: '%s', error: %+v", gcpServiceAccountPath, err)
+		}
+		gcpServiceAccountSecrets, err := polarisreporting.GetPolarisReportingSecrets(namespace, gcpServiceAccountData)
+		if err != nil {
+			return fmt.Errorf("failed to create GCP Service Account Secrets: %+v", err)
 		}
 
-		var objectArr []interface{}
-		for _, v := range components {
-			objectArr = append(objectArr, v)
+		// Print the Secret
+		for _, obj := range gcpServiceAccountSecrets {
+			PrintComponent(obj, "YAML") // helm only supports yaml
 		}
 
-		PrintComponents(objectArr, nativeFormat)
+		// Print Polaris-Reporting Resources
+		err = util.TemplateWithHelm3(polarisReportingName, namespace, polarisReportingChartRepository, helmValuesMap)
```

Comment:
This needs to be changed to be for Polaris :)

Reply:
done